File: test/tile_matrix_test.dart
import 'package:bloo_dot_evolutions/algo/tile_matrix.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  group('algo::matrix', () {
    test("When_CoordinatesSet_Expect_RetrieveSame", () async {
      var testMatrix = TileMatrix(3);
      testMatrix.setCoordinates(2, 1);

      expect(testMatrix.getX(), 2);
      expect(testMatrix.getY(), 1);

      testMatrix.clear();

      expect(testMatrix.getX(), -1);
      expect(testMatrix.getY(), -1);
    });

    test("When_RotateOnce_Expect_RotatedOnce", () async {
      var testMatrix = TileMatrix(3);
      testMatrix.setCoordinates(2, 1);

      testMatrix.rotate();

      expect(testMatrix.getX(), 1);
      expect(testMatrix.getY(), 2);
    });
  });
}

File: app/src/test/java/oy/sarjakuvat/flamingin/bde/algo/TileMatrixTest.kt
package oy.sarjakuvat.flamingin.bde.algo

import org.junit.Test

import org.junit.Assert.*

class TileMatrixTest {
    @Test
    fun when_coordinatesSet_expect_retrieveSame() {
        val testMatrix = TileMatrix(3)
        testMatrix.setCoordinates(2, 1)

        assertEquals(2, testMatrix.getX())
        assertEquals(1, testMatrix.getY())

        testMatrix.clear()

        assertEquals(-1, testMatrix.getX())
        assertEquals(-1, testMatrix.getY())
    }

    @Test
    fun when_rotateOnce_expect_rotatedOnce() {
        val testMatrix = TileMatrix(3)
        testMatrix.setCoordinates(2, 1)

        testMatrix.rotate()

        assertEquals(1, testMatrix.getX())
        assertEquals(2, testMatrix.getY())
    }
}
